### DISCUSSION TITLE
multer not uploads on deploy, but work on localhost (Next.js)

### DISCUSSION THREAD

**Discussion by Xazgl** [vercel/community#2507] [Help] [answered]
### Summary

Find the reason why I can't save files on the server (But localhost saves ) images to the /public/uploads folder. Uploads exists and its access rights are drwxrwxrwx. There are no errors, the api writes that everything is successful, it saves the file name to the database, but there are no files in the /public/uploads folder itself
Please, help me!
### Additional information

https://stackoverflow.com/questions/76278691/why-multer-not-save-file-on-server-nginx-next-js

**Comment by amyegan** [comment, marked as answer]
You won't be able to upload new files to a live Vercel Project outside of build and deployment time. But you can use an external service to store files uploaded via your site.

Related docs and guides:
- [How to Upload and Store Files with Vercel](https://vercel.com/guides/how-to-upload-and-store-files-with-vercel)
- [How can I use files in Serverless Functions on Vercel?](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions)
- [Vercel Blob](https://vercel.com/docs/storage/vercel-blob#) _(currently in private beta)_

**Reply by Xazgl** [reply]
Is there a way to save it to the  my server with Next  simply , but to mine in any way , and then , for example , I run the build ?  Are there any workarounds, save the file to the server, then just call the build of the project and so that it would work?

**Reply by amyegan** [reply]
There are many ways to handle file uploads. It's possible to use the GitHub API to [create or update file contents](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents) and kick off a new deployment from the new commit. A [CMS](https://vercel.com/integrations#cms) is another good option. The right solution depends on your project needs and what tools you are comfortable using.

**Reply by Xazgl** [reply]
My needs are to add an image through the administration panel, and then you can just run the build project again to display it, so I want to find the easiest way to implement it into the project with minimal risks of incompatibility and redoing.
                                
P.S.          I see you are well versed in Vercel, can you please help me in this topic ?)       https://github.com/vercel/next.js/discussions/50028

**Reply by ttsoares** [reply]
As my app using Prisma/Postgres is not working in Vercel... Could I change the process of data storage by writing JSONs files in the Vercel side file system ?  In other world, instead of uploading files - as here - my app will be able to write content using 'fs' ?
